frb_example/with_flutter/lib/bridge_generated.io.dart

```diff
@@ -12,7 +12,26 @@ import 'package:meta/meta.dart';
 import 'dart:ffi' as ffi;
 
 class FlutterRustBridgeExamplePlatform extends FlutterRustBridgeBase<FlutterRustBridgeExampleWire> {
-  FlutterRustBridgeExamplePlatform(ffi.DynamicLibrary dylib) : super(FlutterRustBridgeExampleWire(dylib));
+  final _port = RawReceivePort();
+  NativePortType get port => _port.sendPort.nativePort;
+  FlutterRustBridgeExamplePlatform(ffi.DynamicLibrary dylib) : super(FlutterRustBridgeExampleWire(dylib)) {
+    var initResult =
+        dylib.lookupFunction<ffi.IntPtr Function(ffi.Pointer<ffi.Void>), int Function(ffi.Pointer<ffi.Void>)>(
+            'init_dart_api_dl')(ffi.NativeApi.initializeApiDLData);
+
+    if (initResult != 0) {
+      throw 'Failed to initialize Dart API. Code: $initResult';
+    }
+
+    _port.handler = (response) {
+      inner.drop_DartOpaqueFlutterRustBridgeExample(response);
+    };
+    dylib.lookupFunction<ffi.IntPtr Function(ffi.Pointer<ffi.Void>), int Function(ffi.Pointer<ffi.Void>)>(
+        'init_dart_api_dl')(ffi.NativeApi.initializeApiDLData);
+  }
+  void close() {
+    _port.close();
```

Comment:
⚠️ Problem: What if an object is dropped (and thus send object to this port) *after* someone calls this `close` method? Then it looks never be dropped and memory leak.

Reply:
For this case, we can catch drop errors (writing to a closed port) and save the necessary pointers in `lazy_static`, after which, on the Dart side, clean up if an error occurs.

(The first thing that came to mind)

Reply:
Hmm looks quite complicated, not very sure whether this is the correct way to do so

Reply:
1. how about making close() (port) asynchronous, which will wait for the execution of incomplete functions and then we will throw an error when trying to use the api

2. we can omit ports altogether and use the above mechanism, but then DartOpaque will be removed after the function is executed (for thread_pool)

Reply:
> which will wait for the execution of incomplete functions and then we will throw an error when trying to use the api

IMHO the DartOpaque object can live for arbitrarily long time, because it can be passed to other threads etc, or even assign it to a static variable

> we can omit ports altogether and use the above mechanism, but then DartOpaque will be removed after the function is executed (for thread_pool)

That does not look quite good, since users may want to save DartOpaque somewhere and thus outlive the function

Reply:
Given that Flutter users may never `close` it at all, maybe we (almost) do not need to change your original code at all: Maybe we should simply warn the user if "object dropped after close", possibly by a `log::warn!` macro?

This is imperfect solution as well indeed.

Reply:
in the second case, we will not delete all DartOpaque, but only for which Drop

Reply:
> Maybe we should simply warn the user if "object dropped after close", possibly by a log::warn! macro?
In this case, memory leaks will occur.

Reply:
> in the second case, we will not delete all DartOpaque, but only for which Drop

Not exactly sure the details of the second case indeed...

> In this case, memory leaks will occur.

Yes, that's exactly what I said in "⚠️ Problem" (i.e. first comment). But after rethinking it, I guess Flutter users will *never* close it, so it is not a problem. In addition, we do warn it, so dev will know something bad (but not super critical) is happening.

Reply:
what about this:

give a log about the drop when the port is closed.
save statically leaked pointers and return an error with leaked pointers during the current (next) api call. on the frb side, we clear the memory and return the error further (without pointers)

thus there will be no errors if the api is closed and the api is used without dart opaque.
memory will not leak. what will change - statically store leaked pointers and add handling of such an error on the frb side

Reply:
> save statically leaked pointers and return an error with leaked pointers during the current (next) api call. on the frb side, we clear the memory and return the error further (without pointers)

A slight paraphrase, hopefully making the idea clearer: during any api call, make an extra check whether "leaked pointers" pool is non-empty. If so, trigger cleaning procedure, since we are on the main thread and can safely perform the operation.

Reply:
One worry: we need to have one more check (maybe atomic read) for each and every frb call.

But it seems not to be a problem, since we may be able to make DartOpaque an optional feature if someone thinks it is too slow. Anyway the slowness may not appear *now* at all, since the current implementation is sub-optimal and there should exist many other places which can be improved firstly. So no worries.

Reply:
>A slight paraphrase, hopefully making the idea clearer: during any api call, make an extra check whether "leaked pointers" pool is non-empty. If so, trigger cleaning procedure, since we are on the main thread and can safely perform the operation.

in general, yes, but the idea was precisely at the end of the call, because by this moment all dartOpaque will be deleted (except for the exotic case when we gave dartOpaque to another thread inside the api execution)

Reply:
> except for the exotic case 

IMHO we cannot detect this and this can happen

Reply:
but that's okay - the next call will return an error and clear the memory.
Well, if there is no next call, the logs will report that there was an unsuccessful cleaning attempt.

Reply:
Seems OK, but as I do not have much brain power for this currently (quite busy reading papers), I cannot say this method LGTM now. Anyway good discussions!

Reply:
I think really it is possible to give only warning log.